doc/user/sql/create-source/avro-file.md

```diff
@@ -0,0 +1,145 @@
+
+#### `WITH` options
+
+The following options are valid within the `WITH` clause.
+
+Field | Value | Description
+------|-------|------------
+`tail` | `bool` | Continually check the file for new content; as new content arrives, process it using other `WITH` options.
+
+## Details
+
+This document assumes you're sending Avro-encoded data to a local file, which
+you want Materialize to consume.
+
+### File source details
+
+`path` values must be the file's absolute path, e.g.
+
+```sql
+CREATE SOURCE server_source FROM FILE '/Users/sean/server.log'...
+```
+
+### Avro OCF format details
+
+Materialize will structure data from the file using the Avro OCF's embedded
+schema. For more details, see [Apache Avro 1.9.2 Specification: Object Container
+Files](https://avro.apache.org/docs/1.9.2/spec.html#Object+Container+Files).
+
+### Envelope details
+
+Envelopes determine whether an incoming record inserts new data, updates or
+deletes existing data, or both. For more information, see [API Components:
+Envelopes](../../../overview/api-components#envelopes).
+
+Avro-encoded data is the only type of data that can support the Debezium
+envelope.
+
+### Debezium envelope details
+
+The Debezium envelope provides a "diff envelope", which describes the decoded
+records' old and new values; this is roughly equivalent to the notion of Change
+Data Capture, or CDC. Materialize can use the data in this diff envelope to
+process data as representing inserts, updates, or deletes.
+
+This envelope is called the Debezium envelope because it's been developed to
+explicitly work with [Debezium].
+
+To use the Debezium envelope with Materialize, you must configure Debezium with
+your database.
+
+- [MySQL](https://debezium.io/documentation/reference/0.10/connectors/mysql.html)
+- [PostgreSQL](https://debezium.io/documentation/reference/0.10/connectors/postgresql.html)
+
+The Debezium envelope is only supported by sources published to Kafka by
+Debezium.
+
+#### Format implications
+
+Using the Debezium envelopes changes the schema of your Avro-encoded Kafka
+topics to include something akin to the following field:
+
+```json
+{
+    "type": "record",
+    "name": "envelope",
+    "fields": [
+        {
+        "name": "before",
+        "type": [
+            {
+            "name": "row",
+            "type": "record",
+            "fields": [
+                {"name": "a", "type": "long"},
+                {"name": "b", "type": "long"}
+            ]
+            },
+            "null"
+        ]
+        },
+        { "name": "after", "type": ["row", "null"] }
+    ]
+}
+```
+
+Note that the following section depends on the column's names and types, and is
+unlikely to match our example:
+
+```json
+...
+"fields": [
+        {"name": "a", "type": "long"},
+        {"name": "b", "type": "long"}
+    ]
+...
+```
+
+## Examples
+
+```sql
+CREATE SOURCE events
+FROM FILE '[path to .ocf]'
+WITH (tail = true)
+FORMAT AVRO OCF;
```

Comment:
Might be worth including `ENVELOPE NONE` here to point out that folks should consider what envelope they're using?